### PR TITLE
Add support for kubernetes version 1.31.0

### PR DESCRIPTION
**Pull request by mauri870**
## Proposed commit message

Add support for kubernetes version 1.31.0 released 13 Aug 2024.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates to https://github.com/elastic/observability-dev/issues/3718

### CODE REVIEW

File: .buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml [hunk 1]
```diff
@@ -46,12 +46,12 @@ steps:
...
-            context: "deploy/k8s test v1.30.2"
+            context: "deploy/k8s test v1.31.0"
 
-    - label: "K8S Test/K8S version: v1.29.4"
-      key: "k8s-test-129"
+    - label: "K8S Test/K8S version: v1.30.2"
+      key: "k8s-test-130"
       env:
-        K8S_VERSION: "v1.29.4"
+        K8S_VERSION: "v1.30.2"
```
Comment (gizas) [inline .buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml:54]: 1.30.4 is the latest available I see in https://hub.docker.com/r/kindest/node/tags

File: deploy/kubernetes/Jenkinsfile.yml [hunk 2]
```diff
@@ -18,5 +18,5 @@ stages:
             make check-no-changes;
         stage: checks
     k8sTest:
-        k8sTest: "v1.30.0,v1.29.4,v1.28.9,v1.27.13"
+        k8sTest: "v1.31.0,v1.30.0,v1.29.4,v1.28.9"
```
Comment (gizas) [inline deploy/kubernetes/Jenkinsfile.yml:21]:
```suggestion
        k8sTest: "v1.31.0,v1.30.4,v1.29.8,v1.28.13"
```

File: .buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml [hunk 3]
```diff
@@ -68,12 +68,12 @@ steps:
...
-            context: "deploy/k8s test v1.29.4"
+            context: "deploy/k8s test v1.30.2"
 
-    - label: "K8S Test/K8S version: v1.28.9"
-      key: "k8s-test-128"
+    - label: "K8S Test/K8S version: v1.29.4"
+      key: "k8s-test-129"
       env:
-        K8S_VERSION: "v1.28.9"
+        K8S_VERSION: "v1.29.4"
```
Comment (gizas) [inline .buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml:76]:
```suggestion
        K8S_VERSION: "v1.29.8"
```

File: .buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml [hunk 4]
```diff
@@ -90,12 +90,12 @@ steps:
         machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
       notify:
         - github_commit_status:
-            context: "deploy/k8s test v1.28.9"
+            context: "deploy/k8s test v1.29.4"
 
-    - label: "K8S Test/K8S version: v1.27.13"
-      key: "k8s-test-127"
+    - label: "K8S Test/K8S version: v1.28.9"
```
Comment (gizas) [inline .buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml:95]:
```suggestion
    - label: "K8S Test/K8S version: v1.28.13"
```

File: .buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml (hunk 1 above)
Comment (mauri870) [reply .buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml]: Thanks, I will add the changes in a single commit.

**Review by gizas — APPROVED** [review]

**Review by jlind23 — APPROVED** [review]

**Pull request merged**